SDK/AppCenterCrashes/Microsoft.AppCenter.Crashes.Windows.Shared/Utils/ErrorLogHelper.cs

```diff
@@ -0,0 +1,240 @@
+        public const string ErrorLogFileExtension = ".json";
+
+        /// <summary>
+        /// Error log directory within application files.
+        /// </summary>
+        public const string ErrorStorageDirectoryName = "Microsoft.AppCenter.Error";
+
+        /// <summary>
+        /// Device information utility. Public for testing purposes only.
+        /// </summary>
+        public static IDeviceInformationHelper DeviceInformationHelper;
+
+        /// <summary>
+        /// Process information utility. Public for testing purposes only.
+        /// </summary>
+        public static IProcessInformation ProcessInformation;
+
+        /// <summary>
+        /// File system utility. Public for testing purposes only.
+        /// </summary>
+        public static FileHelper FileHelper;
+
+        /// <summary>
+        /// Static lock object.
+        /// </summary>
+        private readonly static object LockObject = new object();
+
+        static ErrorLogHelper()
+        {
+            DeviceInformationHelper = new DeviceInformationHelper();
+            ProcessInformation = new ProcessInformation();
+            FileHelper = new FileHelper(ErrorStorageDirectoryName);
+        }
+
+        /// <summary>
+        /// Creates an error log for the given exception object.
+        /// </summary>
+        /// <param name="exception">The exception.</param>
+        /// <returns>A new error log instance.</returns>
+        public static ManagedErrorLog CreateErrorLog(System.Exception exception)
+        {
+            return new ManagedErrorLog
+            {
+                Id = Guid.NewGuid(),
+                Timestamp = DateTime.UtcNow,
+                Device = DeviceInformationHelper.GetDeviceInformation(),
+                ProcessId = ProcessInformation.ProcessId ?? 0,
+                ProcessName = ProcessInformation.ProcessName,
+                ParentProcessId = ProcessInformation.ParentProcessId,
+                ParentProcessName = ProcessInformation.ParentProcessName,
+                AppLaunchTimestamp = ProcessInformation.ProcessStartTime,
+                Architecture = ProcessInformation.ProcessArchitecture,
+                Fatal = true,
+                Exception = CreateModelException(exception)
+            };
+        }
+
+        /// <summary>
+        /// Gets all files with the error log file extension in the error directory.
+        /// </summary>
+        public static IEnumerable<FileInfo> GetErrorLogFiles()
+        {
+            lock (LockObject)
+            {
+                try
+                {
+                    // Convert to list so enumeration does not occur outside the lock.
+                    return FileHelper.EnumerateFiles($"*{ErrorLogFileExtension}").ToList();
+                }
+                catch (System.Exception ex)
+                {
+                    AppCenterLog.Error(Crashes.LogTag, "Failed to retrieve error log files.", ex);
+                }
+                return null;
```

Comment:
Should return empty list

```suggestion
                return new List<FileInfo>();
```

Reply:
Looks like addressed in the other PR.